model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2552,14 +2595,9 @@ async def execute(
             request.model_cfg.checkpoint_path,
             is_batch_job=True,
         )
-        # Reconcile gpus count with num_shards from request
         assert hardware.gpus is not None
-        if request.model_cfg.num_shards:
```

Comment:
did this get moved somewhere? otherwise do we want to remove it? (did we get rid of num_shards or something?)

Reply:
I think we should deprecate it; we should just rely on our hardware estimation but i'm open to leaving it

Reply:
@yunfeng-scale do you know of any reason to specify num_shards directly? otherwise I think we should deprecate it too

Reply:
I guess if we're deprecating it we're still gonna need to set `engine_request.model_cfg.num_shards = hardware.gpus`